[Yao Jiaxin] iP

yaojiax opened the pull request:
# Duke Atreides
> "I must not fear. Fear is the mind-killer. Fear is the little death that brings total obliteration. I will face my fear. I will permit it to pass over me and through me. And when it has gone past I will turn the inner eye to see its path. Where the fear has gone there will be nothing. Only I will remain." - Frank Herbert [(source)](https://www.goodreads.com/work/quotes/3634639-dune)

Fearing all your deadlines? Or rather, do you fear forgetting these deadlines? Then fear not, for Duke Atreides of Caladan is here! It can help you look remember your tasks so you do not need to fear forgetting them again, because Duke Atreides is 
- text-based
- easy-to-laern
- ~~evil like the Harkonnens~~ *SUPER FAST* to use

All you need to do is to 
1. download it from [here](https://github.com/yaojiax/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 👯🕺 

And it is **FREE**!

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Here is the main method for those back in Caladan 
```Java
public static void main(String[] args) {
    new Duke().run();
}
```

Commit:
Add Gradle support

yaojiax committed:
implement functionality for greeting, echoing and exiting.

yaojiax committed:
implement functionality for greeting, echoing and exiting.

yaojiax committed:
implement functionality for adding to a list and displaying the list.

yaojiax committed:
implement functionality for marking as done.

yaojiax committed:
implement functionality for ToDo, Events, Deadlines. Separated the classes into different files.

yaojiax committed:
Implemented regression testing

yaojiax committed:
Implemented error handling. Created new classes for the error classes

yaojiax committed:
Implemented delete function.

yaojiax committed:
Implement saving into a text file.

yaojiax committed:
Fix a small bug for parsing deadline commands in the text file

yaojiax committed:
Add the ability to parse Date for Deadline Tasks. Inputs must be of the format YYYY-MM-DD for Deadline.

yaojiax committed:
Merge branch 'branch-Level-7'

yaojiax committed:
Merge branch 'branch-level-8'

# Conflicts:
#	src/main/java/Duke.java

yaojiax committed:
Fix a bug regarding the formatting of date for deadline

yaojiax committed:
Merge remote-tracking branch 'origin/add-gradle-support'

yaojiax committed:
Add more OOP

yaojiax committed:
Divide the classes into packages

yaojiax committed:
Implement Gradle support

yaojiax committed:
Add JUnit Tests

yaojiax committed:
Tweak code to comply with coding standards

yaojiax committed:
Add JavaDocs comments for slightly over half of the methods.

yaojiax committed:
Merge branch 'branch-A-JavaDoc'

yaojiax committed:
Merge branch 'branch-A-CodingStandard'

yaojiax committed:
Implement find function to find matching tasks.

yaojiax committed:
Merge branch 'branch-level-9'

yitong241 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,128 @@
+                validateCmd(arrOfStr[1]);
+            } catch (MissingDescriptionException e) {
+                System.out.println(e.getMessage());
+            }
```
Will it be better if to add indentation after switch?

yitong241 commented on src/main/java/duke/tasklist/Task.java:
```diff
@@ -0,0 +1,53 @@
+     * @return String representation of Task.
+     */
+    public String toString() {
+        String var10000 = this.getStatusIcon();
```
Should the varible name 'var10000' be changed so reader can better understand?

yitong241 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+            this.ui.sayBye();
+        }
+        storage.saveToFile(tasks);
+        //...
```
Maybe the dead code shoule be deleted?

yitong241 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,144 @@
+ * A storage instance handles all interactions between local data file and Duke.
+ */
+public class Storage {
+    private File myFoo;
```
Maybe using od 'myFile' instead of 'myFoo' is better?

yitong241 commented on src/test/java/duke/TodoTest.java:
```diff
@@ -0,0 +1,17 @@
+public class TodoTest {
+
+    @Test
+    public void testToStringComplete() {
+        Todo task = new Todo("This is Todo.");
+        task.changeCompletion();
+        String actualOutput = task.toString();
+        assertEquals("[T][X] This is Todo.", actualOutput);
```
I like how you use the functions. Good job!

yitong241 reviewed (commented):
The code is easy to read and understand! Well done! Just there are a few minor coding quality violations.

yaojiax committed:
Implement GUI for Duke

yaojiax committed:
Implement GUI for Duke

yaojiax committed:
Merge branch 'branch-level-10'

Zeli0 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,144 @@
+     */
+    public void saveToFile(TaskList itemList) {
+        String path = "data/duke.txt";
+        FileWriter fooWriter = null; //init
```
Building on from before, maybe this can also become a fileWriter instead of a fooWriter?

Zeli0 commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,65 @@
+    public TaskList() {
+        this.arrList = new ArrayList<Task>();
+    }
+    public int size() {
```
Should this be changed to getSize() so that the method name is a verb?

Zeli0 commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,65 @@
+
+import java.util.ArrayList;
+
+public class TaskList {
```
Remember to add comments to the class and methods in this file!

Zeli0 reviewed (commented):
Overall your code looks great! I just have a few minor suggestions.

Zeli0 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,138 @@
+            } catch (MissingDescriptionException e) {
+                System.out.println(e.getMessage());
+            }
+            switch (arrOfStr[0]) {
```
Your switch is really neatly organised!

Zeli0 commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,65 @@
+        this.arrList.set(index-1, toMark);
+    }
+    public void setToUnmark(int index) {
+        Task toMark = arrList.get(index - 1);
```
Maybe this can be changed into a better noun? I kind of thought that with the existence of verb methods like toString, toMark would also be considered as a verb, but I can also see that toMark can serve as a noun, so maybe keeping it as it is is good.

Zeli0 reviewed (commented):
Sorry, just added a few extra comments.

yaojiax committed:
Add asserts in code to verify correctness in code

yaojiax committed:
Add assert statements to code

There are currently no assert statements in the code to test for correctness of the program during execution.

Adding asserts help to test the program correctness and avoid bugs on runtime.

As a step forward, we can add more asserts in future feature implementations.

yaojiax committed:
Refactor code to align with coding quality

Parser method was too length and violated SLAP.

Abstracting away the details into new classes help to make code more readable.

Future implementations follow SLAP principle to make the code readable and easy to build on.

yaojiax committed:
Merge pull request #1 from yaojiax/branch-A-CodeQuality

Refactor code to align with coding quality

yaojiax committed:
Merge pull request #2 from yaojiax/branch-A-Assertions

Branch a assertions

yaojiax committed:
Add new feature that can track classes to be done within a time period

Lack support for a task period to be done within a time frame.

Creating this adds more functionality.

Can implement more relevant functions such as do after a certain time period.

yaojiax committed:
Implement changes that allow dynamic resizing of text box.

Future implementation can include dynamic resizing of the window.

yaojiax committed:
Remove dead code

yaojiax committed:
Update README.md

yaojiax committed:
Update README.md

yaojiax committed:
Update README.md

yaojiax committed:
Update README.md

yaojiax committed:
Update README.md

yaojiax committed:
Update README.md

yaojiax committed:
Update README.md

yaojiax committed:
Update README.md

yaojiax committed:
Update README.md

yaojiax committed:
Update README.md

yaojiax committed:
Update README.md

yaojiax committed:
Update README.md

yaojiax committed:
Add a screenshot of the product

yaojiax committed:
Implement a minor visual change to how texts are displayed for list

yaojiax committed:
Merge remote-tracking branch 'origin/master'

yaojiax committed:
Fix minor bugs that were found.
